client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/OperationDispatcher.java

```diff
@@ -268,10 +267,9 @@ private <E> CompletionStage<E> execute(HotRodOperation<E> operation, Set<SocketA
    }
 
    public <E> CompletionStage<E> executeOnSingleAddress(HotRodOperation<E> operation, SocketAddress socketAddress) {
-      if (connectionFailedServers.contains(socketAddress)) {
-         if (log.isTraceEnabled()) {
-            log.tracef("Server %s is suspected, trying another for %s", socketAddress, operation);
-         }
+      // We do an empty check, as contains will perform hashCode on the socketAddress creating a String object
```

Comment:
I guess we should eventually add a SocketAddress subclass to avoid that?

Reply:
Yeah, I kept thinking of how to add a socket wrapper class. But unfortunately, it gets into everything including the ConsistentHash which is a "publicish" API. So I have to go back and see if I can get an acceptable solution.